TicTacToe2/TicTacToeGame.cs

```diff
@@ -24,115 +28,251 @@ public void Play()
         {
-            Console.Clear();
-            Console.WriteLine($"\n\t {space[0]} | {space[1]} | {space[2]}");
-            Console.WriteLine("\t-----------");
-            Console.WriteLine($"\t {space[3]} | {space[4]} | {space[5]}");
-            Console.WriteLine("\t-----------");
-            Console.WriteLine($"\t {space[6]} | {space[7]} | {space[8]}");
+            if (gridNumber == 1)
+            {
+                Console.Clear();
+                Console.WriteLine($"\n\t {space[0]} | {space[1]} | {space[2]}");
+                Console.WriteLine("\t-----------");
+                Console.WriteLine($"\t {space[3]} | {space[4]} | {space[5]}");
+                Console.WriteLine("\t-----------");
+                Console.WriteLine($"\t {space[6]} | {space[7]} | {space[8]}");
+            }
+            else
+            {
+                Console.Clear();
+                Console.WriteLine($"\n\t  {space[0]} | {space[1]} | {space[2]} | {space[3]}");
+                Console.WriteLine("\t--------------------");
+                Console.WriteLine($"\t  {space[4]} | {space[5]} | {space[6]} | {space[7]}");
+                Console.WriteLine("\t--------------------");
+                Console.WriteLine($"\t  {space[8]} | {space[9]} | {space[10]} | {space[11]}");
+                Console.WriteLine("\t--------------------");
+                Console.WriteLine($"\t  {space[12]} | {space[13]} | {space[14]} | {space[15]}");
+            }
+
         }
 
         private void GetUserInput()
         {
             string choiceText;
             bool validInput = false;
-            bool legalMove = false;
 
-            while (!validInput && !legalMove)
+            while (!validInput)
             {
                 Console.WriteLine($"\nIt is player {player}'s turn.");
                 Console.Write("Pick a number from the board to place your mark: ");
                 choiceText = Console.ReadLine();
 
-                bool isParsable = Int32.TryParse(choiceText, out int choice);
-                //choice--;
-                if (!isParsable)
+                bool isParsable = Int32.TryParse(choiceText, out choice);
+                
+                if (!isParsable && !correctNumber)
                 {
-                    validInput = false;
                     Console.WriteLine("Please enter a number.");
+                    validInput = false;
                 }
-                else if (choice <= 0 || choice > 9)
+                else if (correctNumber = CorrectChoice())
                 {
+                    Console.WriteLine("Enter a correct number.");
                     validInput = false;
-                    Console.WriteLine("Please enter a number between 1-9.");
                 }
-                else if (space[choice - 1] == "X" || space[choice - 1] == "O")
+                else if (space[choice] == "X" || space[choice] == "O")
                 {
                     Console.WriteLine($"\n{choice++} is already taken.");
```

Comment:
I didn't catch this before, but there are a couple of problems with this.

The ++ operator has different behavior depending on where it's used.
```c#
choice++ // This increments the value after the variable is used
```
```c#
++choice // This increments the value before the variable is used
```
If you run the program and try to place a second piece on space 1, the program will say that 0 is already taken. This is because the variable is being incremented _after_ it's being used in the statement. Changing to ++choice would show 1, but we don't really want to increment the choice variable here at all. choice + 1 would probably the best option - it doesn't change the variable's value.

![image](https://user-images.githubusercontent.com/47361716/174899307-1d8ed0ed-e184-45de-9dfb-1a89d3541223.png)